instance/models/mixins/storage.py

```diff
@@ -232,14 +232,23 @@ def get_s3_connection(self):
             self.s3_secret_access_key
         )
 
-    def _create_bucket(self, attempts=4, ongoing_attempt=1, retry_delay=4):
+    def _create_bucket(self, attempts=4, ongoing_attempt=1, retry_delay=4, location=None):
         """
         Create bucket, retry up to defined attempts if it fails
+        If you specify a location (e.g. 'EU', 'us-west-1'), it will use it.
         """
         time.sleep(retry_delay)
+        location = location or self.s3_region
         try:
             s3 = self.get_s3_connection()
-            bucket = s3.create_bucket(self.s3_bucket_name)
+            try:
+                bucket = s3.create_bucket(self.s3_bucket_name, location=location)
```

Comment:
@lgp171188 Will this part of the code also need to include special handling for specific regions?

> Different zones have different settings, e.g. Paris [requires AWSv4 signatures](https://stackoverflow.com/questions/27400105/using-boto-for-aws-s3-buckets-for-signature-v4).

Reply:
@itsjeyd, I am not sure if `boto` handles that transparently. Will check this on staging and post an update.

Reply:
@lgp171188 OK, sounds good.